widget/embedded/src/lib/tabManager/tabManager.ts

```diff
@@ -0,0 +1,235 @@
+import type {
+  Events,
+  ID,
+  Payload,
+  TabManagerInterface,
+} from './tabManager.types';
+
+import { CHANNEL_NAME, TIMEOUT } from './tabManager.constants';
+
+export class TabManager implements TabManagerInterface {
+  private id: ID;
+  private claimed: boolean | null = null;
+  private lastCheck: number | undefined;
```

Comment:
I think we need a better name for the last check. It doesn't communicate well the purpose of this variable and it took me a time to understand this variable purpose.